js/components/main.js

```diff
@@ -482,37 +483,46 @@ class Main extends ImmutableComponent {
     currentWindow.on('maximize', function () {
       windowActions.setMaximizeState(true)
     })
-
     currentWindow.on('unmaximize', function () {
       windowActions.setMaximizeState(false)
     })
-
+    // NOTE: setTimeout is used to avoid spamming events
+    let resizeTimeout = null
     currentWindow.on('resize', function (event) {
-      windowActions.saveSize(event.sender.getSize())
+      if (resizeTimeout) {
+        clearTimeout(resizeTimeout)
+      }
+      resizeTimeout = setTimeout(function () {
+        const size = event.sender.getSize()
+        // NOTE: the default window size is whatever the last window resize was
+        appActions.defaultWindowParamsChanged(size)
```

Comment:
be careful with these inside events. This is ok because it's wrapped in a setTimeout (see below for possible change to debounce which would also be ok). Otherwise they should be wrapped in `setImmediate`. No change needed for that, just a reminder.